File: media/textsamples/sample.cs
// sample.cs — C# syntax-highlighting sample for the UltraCanvas demo.
using System;
using System.Collections.Generic;
using System.Linq;

namespace UltraCanvas.Demo
{
    public interface IShape
    {
        double Area();
        string Name { get; }
    }

    public sealed class Circle : IShape
    {
        private readonly double _radius;

        public Circle(double radius) => _radius = radius;

        public string Name => "circle";
        public double Area() => Math.PI * _radius * _radius;
    }

    public static class Program
    {
        public static void Main()
        {
            var shapes = new List<IShape>
            {
                new Circle(1.0),
                new Circle(2.5),
            };

            double total = shapes.Sum(s => s.Area());
            foreach (var s in shapes)
                Console.WriteLine($"{s.Name} -> {s.Area():F4}");

            Console.WriteLine($"total = {total:F4}");
        }
    }
}

File: media/textsamples/sample.rs
// sample.rs — Rust syntax-highlighting sample for the UltraCanvas demo.
use std::f64::consts::PI;

trait Shape {
    fn area(&self) -> f64;
    fn name(&self) -> &str;
}

struct Circle {
    radius: f64,
}

impl Shape for Circle {
    fn area(&self) -> f64 {
        PI * self.radius * self.radius
    }

    fn name(&self) -> &str {
        "circle"
    }
}

fn total_area(shapes: &[Box<dyn Shape>]) -> f64 {
    shapes.iter().map(|s| s.area()).sum()
}

fn main() {
    let shapes: Vec<Box<dyn Shape>> = vec![
        Box::new(Circle { radius: 1.0 }),
        Box::new(Circle { radius: 2.5 }),
    ];

    for shape in &shapes {
        println!("{} -> {:.4}", shape.name(), shape.area());
    }
    println!("total = {:.4}", total_area(&shapes));
}
